src/WebJobs.Script/OutOfProc/Rpc/RpcInitializationService.cs

```diff
@@ -65,8 +65,9 @@ public RpcInitializationService(IOptionsMonitor<ScriptApplicationHostOptions> ap
 
         public async Task StartAsync(CancellationToken cancellationToken)
         {
-            if (Utility.CheckAppOffline(_applicationHostOptions.CurrentValue.ScriptPath))
+            if (Utility.CheckAppOffline(_environment, _applicationHostOptions.CurrentValue.ScriptPath))
```

Comment:
Add separate checks, AppOffline or ContainerDisabled

Reply:
We will have logs to know why the app was offlined. And also prefer to keep it this way so there is just 1 method to check when deciding to do something for offline apps